src/GraphQL/Resolver/QueryType.php

```diff
@@ -266,7 +269,8 @@ public function resolveObjectGetter($value = null, $args = [], $context, Resolve
         $objectList->setUnpublished(1);
         $objectList = $objectList->load();
         if (!$objectList) {
-            throw new ClientSafeException('object with ID ' . $args["id"] . ' not found');
+            $identifiers = ($args["id"] ? ' ID: ' . $args["id"] : '') . ($args["fullpath"] ? ' FULLPATH: ' . $args["fullpath"] : '');
+            throw new ClientSafeException('object with' . $identifiers . ' not found');
         }
         $object = $objectList[0];
```

Comment:
@weisswurstkanone From what I tested so far line 237-275 can be replaced by one line of code calling the new trait:
`$object = $this->getElementByTypeAndIdOrPath($args, 'object');`

I am not sure about edge cases (e.g. setUnpublishd(1) on line 269) and I did not test it with Pimcore 5.7 yet.

Reply:
I don't think so because for objects there is this global sql conditional (a basic condition you can configure)
See https://github.com/pimcore/data-hub/blob/3ff7ada79b3d395911684e4627bec10af71f5bcb/src/GraphQL/Resolver/QueryType.php#L247

Reply:
Right, then I will leave it as is.